src/Microsoft.VisualStudio.Composition/LazyMetadataWrapper.cs

```diff
@@ -195,6 +197,32 @@ public bool Remove(KeyValuePair<string, object> item)
             return metadata;
         }
 
+        internal static bool TryGetMetadataValueTypeRef(IReadOnlyDictionary<string, object> metadata, string key, Resolver resolver, out object typeRef)
+        {
+            typeRef = null;
+
+            if (metadata is LazyMetadataWrapper lazyMetadata)
+            {
+                metadata = lazyMetadata.underlyingMetadata;
+            }
+
+            if (!metadata.TryGetValue(key, out var value))
```

Comment:
>metadata [](start = 17, length = 8)

You're dereferencing this argument without validating that it isn't null coming in from the caller.